bodhi/tests/server/consumers/test_greenwave.py

```diff
@@ -0,0 +1,106 @@
+
+    def setUp(self):
+        super().setUp()
+        self.sample_message = Message(
+            topic="org.fedoraproject.prod.greenwave.decision.update",
+            body={
+                "msg": {
+                    "subject_identifier": "bodhi-2.0-1.fc17",
+                }
+            },
+        )
+        self.handler = greenwave.GreenwaveHandler()
+
+    @mock.patch.dict(config, [('greenwave_api_url', 'http://domain.local')])
+    def test_update_test_gating_status(self):
+        """
+        Assert that a greenwave message updates the gating tests status of an update.
+        """
+
+        self.handler.db_factory = TransactionalSessionMaker(self.Session)
+        update = self.db.query(models.Update).filter(
+            models.Build.nvr == 'bodhi-2.0-1.fc17').one()
+
+        # before the greenwave consumer run the gating tests status is None
+        assert update.test_gating_status is None
+
+        with mock.patch('bodhi.server.models.util.greenwave_api_post') as mock_greenwave:
+            greenwave_response = {
+                'policies_satisfied': True,
+                'summary': "all tests have passed"
+            }
+            mock_greenwave.return_value = greenwave_response
+            self.handler(self.sample_message)
+
+        # After the consumer run the gating tests status was updated.
+        assert update.test_gating_status == models.TestGatingStatus.passed
+
+    @mock.patch('bodhi.server.consumers.greenwave.log')
+    def test_greenwave_bad_message(self, mock_log):
+        """ Assert that the consumer ignores messages badly formed """
+
+        bad_message = Message(topic="", body={})
+        self.handler(bad_message)
+        self.assertEqual(mock_log.debug.call_count, 1)
+        mock_log.debug.assert_called_with("Ignoring message without 'msg'.")
+
+    def test_greenwave_message_missing_info(self):
+        """
+        Assert that the consumer raise an exception if we could not find the
+        subject_identifier in the message
+        """
+
+        bad_message = Message(topic="", body={"msg": {}})
+        with self.assertRaises(BodhiException) as context:
+            self.handler(bad_message)
+
+        self.assertTrue(
+            "Couldn't find subject_identifier in the Greenwave message" in context.exception.args
+        )
+
+    def test_greenwave_wrong_build_nvr(self):
+        """
+        Assert that the consumer raise an exception if we could not find the
+        subject_identifier (build nvr) in the DB.
+        """
+        self.handler.db_factory = TransactionalSessionMaker(self.Session)
+        self.sample_message.body["msg"]["subject_identifier"] = "notapackage-2.0-1.fc17"
+
+        with self.assertRaises(BodhiException) as context:
+            self.handler(self.sample_message)
+
+        self.assertTrue(
+            "Couldn't find build notapackage-2.0-1.fc17 in DB" in context.exception.args
+        )
```

Comment:
The test class is derived from the `unittest.TestCase`, so you can also use `self.assertIn("...", context.exception.args)` here. When my PR #3168 is merged, we can start using test classes not derived from `unittest.TestCase` which, among other goodies, would let you simply use asserts here, i.e. `assert "..." in ...` (but we're not there yet).

Reply:
I'd like to leave this one as it currently is.  It works and I don't see the benifit of switching to asserIn.
Maybe after your code is merged and we can just use assert, we'll change this one.

Reply:
Sure, I just didn't know if you were aware of it.

Reply:
After readying bowlofeggs comment below, and testing it out, it does give a much better debugging message.  I have changed both of my `self.assertTrue` with `self.assertIn`